platforms/gke/base/core/gke_enterprise/configmanagement/git/iam.tf

```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "google_project_iam_member" "otel_collector" {
+  for_each = toset([
+    "roles/monitoring.metricWriter"
```

Comment:
Can we add the role roles/cloudtrace.agent to enable tracing as well?

Reply:
I'm not sure this is really the best place to add that role unless it is a best practice to always apply that role to KSA.

Reply:
I added the role here because it's what I saw related errors in the `otel-collector` logs. `otel-collector` is deployed when enabling Config Sync automatically.

[Google Cloud Managed Service for Prometheus docs](https://cloud.google.com/stackdriver/docs/managed-prometheus/setup-otel#gmp-default-sa) say that the needed permissions are `monitoring.metricWriter` and `monitoring.viewer`, [granted using the `monitoring.metricWriter` role](https://cloud.google.com/stackdriver/docs/managed-prometheus/setup-otel#authorize-sa).

> It seems like it might be better in a separated Cloud Trace or observability Terraservice.

+1. When we have that module, we can also move these `google_project_iam_member` over there maybe?

Reply:
Yeah, I was referring to the `roles/cloudtrace.agent`, the `roles/monitoring.metricWriter` is fine.

Reply:
It would be convenient to have the cloud trace role added. But I am fine to leave it out for now. Actually, I think it's probably better to open a bug for the config sync team to add the `roles/monitoring.metricWriter` role since they installed the `otel-collector`.

Reply:
This is documented in https://cloud.google.com/stackdriver/docs/managed-prometheus/troubleshooting#misconfigured-svcacct. I think Config Sync assumes the default compute engine service account. I'll resolve this one while we think about if we need to implement a dedicated service.